oauth.go

```diff
@@ -156,6 +221,7 @@ func GetKMSProvider(configFilePath string) (vaultName *string, keyName *string,
 // Updates azure.json with key version information
 func UpdateKMSProvider(configFilePath string, keyVersion string) (err error) {
 	var configReader io.Reader
+	var config *Config
```

Comment:
Actually you can unmarshal that as `map[string]json.RawMessage` so you don't have to update the `Config` struct every time upstream has a change. Here is a brief example:
```go
var config map[string]json.RawMessage
if err := json.Unmarshal(configContents, &config); err != nil {
    return err
}

if providerKeyVersionRaw, ok := config["providerKeyVersion"]; ok {
    var providerKeyVersion string
    if err := json.Unmarshal(providerKeyVersionRaw, &providerKeyVersion); err != nil {
        return err
    }

    if !strings.EqualFold(providerKeyVersion, keyVersion) {
        newProviderKeyVersion, err := json.Marshal(keyVersion)
        // Check err

        config["providerKeyVersion"] = json.RawMessage(newProviderKeyVersion)
        newConfig, err := json.MarshalIntent(config, "", "    ")
        // Write to file
    }
}  else {
    // Deal with key not found
}
```

Reply:
@ArchangelSDY thanks for the suggestion! I have updated this PR. Can you PTAL?

Reply:
LGTM. Cheers!